### PR TITLE
Adding a CHANGELOG.md

### PR DESCRIPTION
**Pull request by akshaysubr**
This PR adds a `CHANGELOG.md` based on the format outlined in: https://keepachangelog.com/en/1.0.0/

Also updated the MR template with a checklist item for updating the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Changelog
```
Comment (NickGeneva) [inline CHANGELOG.md:1]: I liked [cuSpatials change log](https://github.com/rapidsai/cuspatial/blob/branch-23.08/CHANGELOG.md#-breaking-changes-1) with the sections (with emotes :100: :sunglasses:  ):

With the points:
- Breaking Changes
- Bug Fixes
- Documentation
- New Features
- Improvements

I like keeping Dependencies (not sure how frequent we have Security updates)

https://docs.rapids.ai/resources/changelog/

Thoughts?

**Comment by ktangsali** [comment]
Looks good to me, can we add a comment in the [`CONTRIBUTING.md`](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md ) to make folks aware of filling this out?

**Comment by akshaysubr** [comment]
@ktangsali Updated the contribution guide to include changelog update instructions

**Review by ktangsali — APPROVED** [review]
LGTM!

**Review by NickGeneva — APPROVED** [review]

**Pull request merged**